api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -615,46 +620,75 @@ func (c *imageBuilder) createKanikoJobSpec(
 		fmt.Sprintf("--dockerfile=%s", baseImageTag.DockerfilePath),
 		fmt.Sprintf("--context=%s", baseImageTag.BuildContextURI),
 		fmt.Sprintf("--build-arg=%s=%s", baseImageEnvKey, baseImageTag.ImageName),
+		fmt.Sprintf("--build-arg=%s=%s", modelArtifactStorageTypeKey, c.artifactService.GetType()),
 		fmt.Sprintf("--build-arg=%s=%s", modelDependenciesUrlEnvKey, modelDependenciesUrl),
 		fmt.Sprintf("--build-arg=%s=%s/model", modelArtifactsUrlEnvKey, version.ArtifactURI),
 		fmt.Sprintf("--destination=%s", imageRef),
 	}
 
+	if c.artifactService.GetType() == "s3" {
+		kanikoArgs = append(
+			kanikoArgs,
+			fmt.Sprintf("--build-arg=%s=%s", "AWS_ACCESS_KEY_ID", os.Getenv("AWS_ACCESS_KEY_ID")),
+			fmt.Sprintf("--build-arg=%s=%s", "AWS_SECRET_ACCESS_KEY", os.Getenv("AWS_SECRET_ACCESS_KEY")),
+			fmt.Sprintf("--build-arg=%s=%s", "AWS_DEFAULT_REGION", os.Getenv("AWS_DEFAULT_REGION")),
+			fmt.Sprintf("--build-arg=%s=%s", "AWS_ENDPOINT_URL", os.Getenv("AWS_ENDPOINT_URL")),
+		)
+	}
 	if c.config.BaseImage.BuildContextSubPath != "" {
 		kanikoArgs = append(kanikoArgs, fmt.Sprintf("--context-sub-path=%s", c.config.BaseImage.BuildContextSubPath))
 	}
 
 	kanikoArgs = append(kanikoArgs, c.config.KanikoAdditionalArgs...)
 
 	activeDeadlineSeconds := int64(c.config.BuildTimeoutDuration / time.Second)
-	var volume []v1.Volume
-	var volumeMount []v1.VolumeMount
+	var volumes []v1.Volume
+	var volumeMounts []v1.VolumeMount
 	var envVar []v1.EnvVar
 
-	// If kaniko service account is not set, use kaniko secret
-	if c.config.KanikoServiceAccount == "" {
-		kanikoArgs = append(kanikoArgs,
-			fmt.Sprintf("--build-arg=%s=%s", gacEnvKey, saFilePath))
-		volume = []v1.Volume{
+	// Configure additional authentication requirements for specific image registries
```

Comment:
Ah do you mean like refactor this block into a separate helper function? Yep that'll make it cleaner :)

Reply:
yes

Reply:
Hmm, I've currently refactored the entire code chunk into tinier helper functions but they are currently grouped by the types of configs/specs that are being modified (instead of the type of push registry):
```go
	// Configure additional authentication requirements for specific image registries
	kanikoArgs = c.configureKanikoArgsForKanikoPushRegistry(kanikoArgs)
	volumes, volumeMounts = c.configureVolumesAndVolumeMountsForKanikoPushRegistry(volumes, volumeMounts)
	envVar = c.configureEnvVarsForKanikoPushRegistry(envVar)
```